### PR TITLE
Function is it right time

### PR DESCRIPTION
**Pull request by pjap93**
function isItRightTime.

Valide if the time is in the range definided.
For the addition in the conditions of the scenarios.

### CODE REVIEW

**Comment by Pierre-Gilles** [comment]
Hey! Is it ready for merge ? :) What's your use case on this ?

**Comment by pjap93** [comment]
Hi,
it is close if there is no error, but it is impossible to understand what is causing problems with Travis lol, even by putting the test file on it.
When I read the details of the error it indicates that the function does not exist while I declare everything (well I think).

I currently use it under scenario conditions to set a validation time range.

I would then make a PR in the data to add the condition if this is ok for you.

**Comment by MathieuAndrade** [comment]
Use `Promise.resolve(true)` and `Promise.resolve(false)` instead of just returning a boolean ;)

**Comment by pjap93** [comment]
is good, thanks

File: api/core/time/time.isItRightTime.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    // end time default is the end of the day
+    var endDate = new Date();
+    endDate.setHours(arrayOfEnd[0] || 23);
+    endDate.setMinutes(arrayOfEnd[1] || 59);
+    endDate.setSeconds(arrayOfEnd[2] || 59);
+
+    // settings default options if not defined
+    options = options || {};
+    options.start = date;
+    options.end = endDate;
```
Comment (Pierre-Gilles) [inline api/core/time/time.isItRightTime.js]: Are these 3 lines useful ? I don't think so!

File: api/core/time/time.isItRightTime.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+module.exports = function isItRightTime(options){
+
+    var now = new Date();
+
+    var arrayOfSart = options.start.split(":");
+    var arrayOfEnd = options.end.split(":");
+
+    // default start time 00:00:00
+    var date = new Date();
```
Comment (Pierre-Gilles) [inline api/core/time/time.isItRightTime.js]: Consider renaming the variable to "startDate", more explicit

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Hey ! Thanks for your PR, this is really nice :) 

I'm merging this just after these two changes in this review

File: api/core/time/time.isItRightTime.js (hunk 1 above)
Comment (pjap93) [reply api/core/time/time.isItRightTime.js]: ah yes well seen, I forget to remove them, initially I wanted to return data after I change a true / false

File: api/core/time/time.isItRightTime.js (hunk 2 above)
Comment (pjap93) [reply api/core/time/time.isItRightTime.js]: it's not my part that I prefer to give explicit names to variables lol
but you already noticed

**Comment by pjap93** [comment]
and in the stride I can add in the base the condition for the scenarios !!

File: api/core/time/time.isItRightTime.js (hunk 1 above)
Comment (Pierre-Gilles) [reply api/core/time/time.isItRightTime.js]: Ok! Waiting for your review before merging :)

Comment (pjap93) [reply api/core/time/time.isItRightTime.js]: it is done :-)

**Comment by Pierre-Gilles** [comment]
It's now merged! :) 

I just renamed the function "isInTimeRange" because your old function name was clearly not explicit. 

Thanks for your work!

**Pull request closed without merging**